### DISCUSSION TITLE
Role based routing with v6.4+ createBrowserRouter

### DISCUSSION THREAD

**Discussion by sealmu** [remix-run/react-router#11136] [Q&A]
How to implement role-based routing with createBrowserRouter? 
I want the same routes(ex. "/", "/profile", "dashboard") depending on the user role logged in to be rendered with the different components(ex. ClientProfile, OwnerProfile, DeliveryProfile). 
And when it's public, for no logged-in user I want only some routes to be enabled(ex. "/" - for MainPage). 

I can create several distinguished routes with createBrowserRouter(ex. ClientRouter, OwnerRouter, etc.) and provide the right one based on the app state to "<RouterProvider router={appRouter}". 
But it looks like a wrong approach and too cumbersome relative to the previous version of the dynamically rendered routes when it could be implemented easily by conditional rendering.

**Comment by kaidouya** [comment]
I also have the same question

**Comment by bhavikcct** [comment]
@sealmu have you find any solution for that ?

**Comment by tatsmaki** [comment]
Routes list can be dynamic, there is nothing wrong in this approach. If you want static routes to be protected you can use some general patterns.

```
const ProfileGuard = () => {
    const { isAuthorized, role } = useAuth();
    
    if (!isAuthorized) {
        return <Navigate to="/auth" />
    }
    
    switch(role) {
        case "delivery": {
            return <DeliveryProfile />;
        }
        case "owner": {
            return <OwnerProfile />;
        } 
        default: {
            return null;
        }
    }
}

const routes = [
    { route: '/auth', element: <Auth /> },
    { route: '/profile', element: <ProfileGuard />
];

const Router = createBrowserRouter(routes);
```
